src/Monitor.vala

```diff
@@ -4,17 +4,41 @@ namespace Monitor {
         private MainWindow window = null;
         public string[] args;
 
-        construct {
-            application_id = "com.github.stsdc.monitor";
+        private static bool start_in_background = false;
+        private static bool status_background = false;
+        private const GLib.OptionEntry[] cmd_options = {
+        // --start-in-background
+            { "start-in-background", 'b', 0, OptionArg.NONE, ref start_in_background, "Start in background with wingpanel indicator", null },
+            // list terminator
+            { null }
+        };
+
+        // contructor replacement, flags added
+        public MonitorApp (bool status_indicator) {
+            Object (
+                application_id : "com.github.stsdc.monitor",
+                flags: ApplicationFlags.FLAGS_NONE
+            );
+            status_background = status_indicator;
         }
 
         public override void activate () {
             // only have one window
-            if (get_windows () == null) {
-                window = new MainWindow (this);
-                window.show_all ();
-            } else {
+            if (get_windows () != null) {
                 window.present ();
+                return;
+            }
+
+            window = new MainWindow (this);
+
+            //start in background with indicator
+            if (status_background) {
+                if (!window.saved_state.indicator_state) {
+                    window.saved_state.indicator_state = true;
+                }
+                window.hide ();
```

Comment:
I think we may want to add new line after closing brackets here. From https://elementary.io/docs/code/reference#whitespace:

> Lines consisting of closing brackets (`}` or `)`) should be followed by an empty 
line, except when followed by another closing bracket or an `else` statement.

Reply:
I personally think the line there breaks the parent `if` code block in an unflattering way. However, changes made to fit the style ;)